crypto/rsa/rsa_oaep.c

```diff
@@ -244,15 +243,16 @@ int RSA_padding_check_PKCS1_OAEP_mgf1(unsigned char *to, int tlen,
      * should be noted that failure is indistinguishable from normal
      * operation if |tlen| is fixed by protocol.
      */
-    tlen = constant_time_select_int(constant_time_lt(dblen, tlen), dblen, tlen);
+    tlen = constant_time_select_int(constant_time_lt(dblen - mdlen - 1, tlen),
+                                    dblen - mdlen - 1, tlen);
     msg_index = constant_time_select_int(good, msg_index, dblen - tlen);
     mlen = dblen - msg_index;
-    for (from = db + msg_index, mask = good, i = 0; i < tlen; i++) {
+    for (mask = good, i = 0; i < tlen; i++) {
         unsigned int equals = constant_time_eq(i, mlen);
 
-        from -= dblen & equals; /* if (i == dblen) rewind   */
+        msg_index -= dblen & equals; /* if (i == dblen) rewind */
```

Comment:
The comment should say i == mlen

Reply:
Yes of course.

Reply:
An alternative would be to make equals check msg_index and dblen, which might be easier to follow.

Reply:
I am not sure I understand this comment:
```
     * bytes are viewed as circular buffer with start at |tlen|-|mlen'|,
     * where |mlen'| is "saturated" |mlen| value
```

seems that applies to the PKCS1 code, where it jumps back to 11,
but this loop jumps back to 0.

Reply:
I could make it jump back to mdlen + 1, where a full size plain text starts

Reply:
The alternative would be easier to understand, and one would avoid head scratching
why the rewind can only happen once.

Reply:
the new version has a very nice property.
db[msg_index++] accesses every byte from db[mdlen+1] to db[dblen - 1] exactly once.